Messaging.md

```diff
@@ -363,90 +374,170 @@ Witnesses should be sorted by the `input_serial_id` sent in `funding_input` defi
+A node *MUST* discard any announcement failing this verification.
+
+#### Rationale
+
+While using an aggregate signature increases the complexity of oracle implementation and management, providing a signature per nonce would lead to a large number of signatures in particular for oracle attesting to numerical event.
+
 ### The `oracle_announcement` Type
 
 This type contains an `oracle_event` and a signature certifying its origination.
 As oracle announcements can be broadcast directly, they are encoded as [wire messages](#wire-messages).
 See [the Oracle specifications](./Oracle.md#oracle-announcements) for more details.
 
-**For backward compatibility reasons, this type is currently serialized as a TLV and uses u16 for collection prefix. It is expected to be changed to the new serialization format in a near future.**
-
 #### `oracle_announcement`
 
-1. type: 55332
-1. data:
-   * [`signature`:`annoucement_signature`]
-   * [`x_point`:`oracle_public_key`]
+1. type: 55354 (`oracle_announcement`)
+2. data:
+   * [`signature`:`announcement_signature`]
+   * [`oracle_metadata`:`oracle_metadata`]
    * [`oracle_event`:`oracle_event`]
 
-where `signature` is a Schnorr signature over a sha256 hash of the serialized `oracle_event`, using the tag `announcement/v0`.
+where both the `announcement_signature` is a Schnorr signature over a sha256 hash of the serialized `oracle_event`, using the tag `announcement/v1`, that should be verified using the `oracle_announcement_public_key`.
 
-### The `oracle_attestation` Type
+#### Requirements
 
-This type contains information about the outcome of an event and the signature(s) over its outcome value(s).
-As oracle attestations can be broadcast directly, they are encoded as [wire messages](#wire-messages).
-See [the Oracle specifications](./Oracle.md#oracle-attestations) for more details.
+Clients SHOULD check the `oracle_attestation_public_key` has been signed by the `oracle_announcement_public_key` in a known `oracle_metadata` message and SHOULD abort if the attestation key is not recognized.
 
-**For backward compatibility reasons, this type is currently serialized as a TLV and uses u16 for collection prefix. It is expected to be changed to the new serialization format in a near future.**
+### The `schnorr_attestation` Type
 
-#### `oracle_attestation`
+The `schnorr_attestation` type provides signatures over an event outcome result values.
 
-1. type: 55400
+1. type: 55356
 1. data:
-    * [`string`:`event_id`]
-    * [`x_point`:`oracle_public_key`]
-    * [`u16`: `nb_signatures`]
-    * [`signature`:`signature_1`]
-    * ...
-    * [`signature`:`signature_n`]
-    * [`string`:`outcome_1`]
-    * ...
-    * [`string`:`outcome_n`]
-
+   * [`string`:`event_id`]
+   * [`x_point`:`oracle_attestation_public_key`]
+   * [`bigsize`: `nb_outcomes`]
+   * [`nb_signatures*attested_outcome`:`attested_outcomes`]
+1. subtype: `attested_outcome`
```

Comment:
So I don't understand why this is necessary. So we use this pattern any where? This complicates things. Why can't we just serialize the signatures / outcomes with bigsize prefixes?